woocommerce/woocommerce-ios
# [Store onboarding] Reload task list on task completed

Closes: #9172


## Description

Reload the onboarding task list upon completing tasks by adding a callback. 

## Testing instructions

Prerequisite: Create a new WPCOM WooCommerce site 

- Launch and log in using a WPCOM site
- Ensure that you see the onboarding list in dashboard
- Tap "Launch your store" and finish the task.
- When you see the "Dashboard" screen, notice that the task list is already reloaded and the "Launch your store" task is marked as completed. 
- Validate the same for "Add your first product" and "Customize your domain" tasks

**Known issues**
- We cannot detect when the "Get paid" task gets completed as that is a web view. So the task list won't be reloaded automatically for that.
- For the "Customize your domain" task, the backend doesn't mark it as completed immediately after purchasing a domain. So, reloading the task list doesn't update the task. Waiting for confirmation - p1679456923870899-slack-C045CUK1Y3U


## Screenshots
![LaunchStore](https://user-images.githubusercontent.com/524475/226795903-ffa77a0d-8d07-4744-ae2e-eaeb0eb82fbf.gif)


---
